File: hw4/clusterer.py
from time import time
import numpy as np
import pylab

from sklearn import metrics
from sklearn.metrics.pairwise import euclidean_distances
from scipy import cluster
X = np.genfromtxt('clusters_input.csv', delimiter=',')
dissE = euclidean_distances(X, X, squared=True)

Z = cluster.hierarchy.linkage(X, method='weighted', metric='minkowski')

c = cluster.hierarchy.fcluster(Z, 9, criterion='maxclust')

pylab.scatter(X[:,0],X[:,1], c=c)
pylab.savefig("minkowski.png")

for i in range(len(c)):
    print(str(c[i]))

File: hw4/clusterer.R
#!/usr/bin/Rscript

library('cluster')

asw <- numeric(20) 
x <- read.csv("clusters_input.csv")

hc <- hclust(dist(x,method='minkowski'))

c <- cutree(hc,9)

for (k in 1:length(c)) {
  cat(c[k],"\n")
}
